packages/metals-vscode/src/testExplorer/testRunHandler.ts

```diff
@@ -110,9 +110,10 @@ export async function runHandler(
         tests: [],
       };
     } else {
+      const { id, _metalsParent } = test;
       return {
-        className: test._metalsParent.id,
-        tests: [test.label],
+        className: _metalsParent.id,
+        tests: [id],
```

Comment:
```suggestion
        className: test._metalsParent.id,
        tests: [test.id],
```
that would be a bit less magical.